v2/.dockerignore

```diff
@@ -0,0 +1,11 @@
+**
+
+# Ignore everything except below
+!/go.mod
+!/go.sum
+!/*.go
+!/internal
+!/pkg
```

Comment:
aren't internal and pkg going to be included because of the `!/*.go` above? Do we really need the called out explicitly?

Reply:
Why does the container need `api` folder?

Reply:
> Why does the container need `api` folder?

Initially I tried without `api` folder and build fails with error below: 
```
[controller:docker-push-multiarch]  > [linux/arm64 builder 7/7] RUN CGO_ENABLED=0 go build -ldflags "-X github.com/Azure/azure-service-operator/v2/internal/version.BuildVersion=v2.3.0-rev.71-dirty" -o ./bin/aso-controller ./cmd/controller/:
[controller:docker-push-multiarch] 1.197 internal/controllers/controller_resources_gen.go:213:2: no required module provides package github.com/Azure/azure-service-operator/v2/api/web/customizations; to add it:
[controller:docker-push-multiarch] 1.197 	go get github.com/Azure/azure-service-operator/v2/api/web/customizations
[controller:docker-push-multiarch] 1.197 internal/controllers/controller_resources_gen.go:214:2: no required module provides package github.com/Azure/azure-service-operator/v2/api/web/v1api20220301; to add it:
[controller:docker-push-multiarch] 1.197 	go get github.com/Azure/azure-service-operator/v2/api/web/v1api20220301
[controller:docker-push-multiarch] 1.197 internal/controllers/controller_resources_gen.go:215:2: no required module provides package github.com/Azure/azure-service-operator/v2/api/web/v1api20220301storage; to add it:
[controller:docker-push-multiarch] 1.197 	go get github.com/Azure/azure-service-operator/v2/api/web/v1api20220301storage
[controller:docker-push-multiarch] 1.197 internal/controllers/controller_resources_gen.go:216:2: no required module provides package github.com/Azure/azure-service-operator/v2/api/web/v1beta20220301; to add it:
[controller:docker-push-multiarch] 1.197 	go get github.com/Azure/azure-service-operator/v2/api/web/v1beta20220301
[controller:docker-push-multiarch] 1.197 internal/controllers/controller_resources_gen.go:217:2: no required module provides package github.com/Azure/azure-service-operator/v2/api/web/v1beta20220301storage; to add it:
[controller:docker-push-multiarch] 1.197 	go get github.com/Azure/azure-service-operator/v2/api/web/v1beta20220301storage
```

Reply:
> aren't internal and pkg going to be included because of the `!/*.go` above? Do we really need the called out explicitly?

Actually, `!/*.go` doesn't make sense, I'll remove it.

Reply:
doh, I mistook `api` for `config`. `api` contains the actual API Go files, which obviously do need to be included.

> Actually, !/*.go doesn't make sense, I'll remove it.

Can you expand on this? Based on my read of dockerignore syntax, `!/*.go` should include any go file (weirdly * seems multi-directory-recursive in .dockerignore even though it's not usually).

Reply:
> !/*.go should include any go file

Is that documented? I thought its just gonna include `.go` files in the same directory as `Dockerfile` is.

Reply:
Ah, you're right. It's docuemnted here: https://docs.docker.com/engine/reference/builder/#dockerignore-file

They support `**` though:
> Beyond Go's filepath.Match rules, Docker also supports a special wildcard string ** that matches any number of directories (including zero). For example, **/*.go will exclude all files that end with .go that are found in all directories, including the root of the build context.

So `!**/*.go` should cover all of the Go files without needing to remember to come update this folder list if we add new directories.